### PR TITLE
add reportedPostsCreate firebase function

### PR DESCRIPTION
**Pull request by jagaSto**
* PR introduces the `reportedBy` property for documents of the `ask-for-help` collection
* with this property, we can track which users reported an entry
* adds firebase function that updates the `reportedBy` on the corresponding `ask-for-help` document, whenever an entry is reported (and therefore added to the `reported-posts` collection)

### CODE REVIEW

File: functions/index.js [hunk 1]
```diff
@@ -225,3 +225,20 @@ exports.regionSubscribeCreate = functions.region('europe-west1').firestore.docum
       console.log('ID', snap.id);
     }
   });
+
+  exports.reportedPostsCreate = functions.region('europe-west1').firestore.document('/reported-posts/{reportRequestId}')
+  .onCreate(async (snap, context) => {
+    try {
+      const db = admin.firestore();
+      const snapValue = snap.val();
```
Comment (kenodressel) [inline functions/index.js]: should be snap.data()

File: functions/index.js [hunk 2]
```diff
@@ -225,3 +225,20 @@ exports.regionSubscribeCreate = functions.region('europe-west1').firestore.docum
...
+      const { postId, uid } = snapValue;
```
Comment (kenodressel) [inline functions/index.js]: could we rename postId to askForHelpId ?

File: functions/index.js (hunk 1 above)
Comment (jagaSto) [reply functions/index.js]: I could only find `.val()` in the documentation, assuming that I was looking at the right [snapshot](https://firebase.google.com/docs/reference/functions/providers_database_.datasnapshot#methods) in the [onCreate](https://firebase.google.com/docs/reference/functions/providers_database_.refbuilder#on-create) function

File: functions/index.js (hunk 2 above)
Comment (jagaSto) [reply functions/index.js]: sure! Will adapt in frontend as well

File: functions/index.js (hunk 1 above)
Comment (jagaSto) [reply functions/index.js]: turns out snap.data() was correct :D

**Comment by jagaSto** [comment]
will open a new branch with cleaner commit history.

**Pull request closed without merging**